core/src/smartcontracts/isi/permissions/has_token.rs

```diff
@@ -64,11 +64,7 @@ impl<H: HasToken> IsAllowed for HasTokenAsValidator<H> {
     ) -> ValidatorVerdict {
         let permission_token = match self.has_token.token(authority, instruction, wsv) {
             Ok(concrete_token) => concrete_token.into(),
-            Err(err) => {
-                return ValidatorVerdict::Deny(format!(
-                    "Unable to identify the corresponding permission token: {err}",
-                ));
-            }
+            Err(_) => return ValidatorVerdict::Skip,
```

Comment:
**IMPORTANT**
This here were, I think, two bugs cancelling each other: validator replied `Deny` instead of `Skip` for inapplicable instructions, but they weren't checked.
I _think_ `HasToken::token` doesn't produce errors that should make validator reply `Deny`, but I am not sure, since it's `Error` doesn't bear any additional context. I think if we merge this, we should open an issue to refactor `HasToken::token` return type and check this line again.

Reply:
did you write a test for confirmation?

Reply:
@mversic All permission tests pass with this PR, but so they do on current master, so our test suite is incomplete either way. I'm investigating the issue currently, won't merge this until I at least know for sure where the bug is.

Reply:
I'd address this in a separate PR, as is, the code here is more important to merge sooner rather than later.